contracts/src/types/LockRequest.sol

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2025 RISC Zero, Inc.
+//
+// All rights reserved.
+pragma solidity ^0.8.20;
+
+import {ProofRequest, ProofRequestLibrary} from "./ProofRequest.sol";
+import {Account} from "./Account.sol";
+import {Callback, CallbackLibrary} from "./Callback.sol";
+import {Offer, OfferLibrary} from "./Offer.sol";
+import {Predicate, PredicateLibrary} from "./Predicate.sol";
+import {Input, InputType, InputLibrary} from "./Input.sol";
+import {Requirements, RequirementsLibrary} from "./Requirements.sol";
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {IBoundlessMarket} from "../IBoundlessMarket.sol";
+import {IERC1271} from "@openzeppelin/contracts/interfaces/IERC1271.sol";
+
+using LockRequestLibrary for LockRequest global;
+
+/// @title Lock Request Struct and Library
+/// @notice Wraps a ProofRequest. Ensures domain separation between client signatures and prover signatures.
```

Comment:
This comment will be more like historical information as time passes. If someone ever asks "why does this exist?" we can provide this information.
```suggestion
/// @notice Message sent by a prover to indicate that they intend to lock the given request.
```